pkg/controller/routes.go

```diff
@@ -104,6 +105,7 @@ func routes(router *router.Router, cfg *rest.Config, registryTransport http.Roun
 	router.Type(&v1.ServiceInstance{}).HandlerFunc(gc.GCOrphans)
 
 	router.Type(&v1.EventInstance{}).HandlerFunc(eventinstance.GCExpired())
+	router.Type(&v1.EventInstance{}).IncludeRemoved().HandlerFunc(eventinstance.Truncate(ctx, router.Backend(), 1000))
```

Comment:
Should I expose these as config parameters?

Reply:
Eventually, probably. I don't think it should block this merging though.